File: 179-largest-number/largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        String[] arr = new String[nums.length];
        for(int i=0; i< nums.length; i++){
            arr[i] = String.valueOf(nums[i]);
        }
        Arrays.sort(arr, (a,b) -> (b+a).compareTo(a+b));
        String result = String.join("", arr);
        return result.charAt(0) == '0' ? "0" : result;
    }
}

File: 179-largest-number/largest-number.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {
    nums = nums.map(String);
    let sorted = nums.sort((a, b) => (b+a)-(a+b));
    let result = nums.join('');
    return result[0] === '0' ? '0' : result;
};